src/opl/chips/opl_serial_port.cpp

```diff
@@ -52,18 +56,23 @@ void OPL_SerialPort::writeReg(uint16_t addr, uint8_t data)
 void OPL_SerialPort::sendSerial(uint addr, uint data)
 {
     QSerialPort *port = m_port;
-
-    // TODO support OPL2 only
-    if(addr >= 0x100)
-        return;
-
-    ///
     if(!port || !port->isOpen())
         return;
 
-    ///
-    uint8_t sendBuffer[2] = {(uint8_t)addr, (uint8_t)data};
-    port->write((char *)sendBuffer, 2);
+    unsigned protocol = m_protocol.load();
+
+    switch(protocol)
+    {
+    // TODO support OPL2 only
+    default:
+    case ProtocolArduinoOPL2:
```

Comment:
Let's ask @nukeykt for his OPL3 rig protocol? The thing would be able to work on his thing also 🤔

Reply:
It's pretty simple:
```
            {
                xx[0 + size] = (regg >> 6) | 0x80;
                xx[1 + size] = ((regg & 0x3f) << 1) | (data >> 7);
                xx[2 + size] = (data & 0x7f);
            }
```
```
  if(Serial.available())
  {
    byte x1 = Serial.read();
    if(x1&0x80)
    {
      while(!Serial.available());
      byte x2= Serial.read();
      while(!Serial.available());
      byte x3 = Serial.read();
      byte reg = ((x1<<6)&0xc0)|((x2>>1)&0x3f);
      byte data = ((x2<<7)&0x80)|(x3&0x7f);
      writereg(reg,data);
    }
  }
```

Reply:
Thanks :fox_face:, and thanks @jpcima for implementation! :fox_face: 
If you want, feel free to check the thing in action if you can :fox_face: :+1:

Reply:
Cool, thanks. Unfortunately my OPL3 rig was prototyped on breadboard and i disassembled it years ago to use breadboard for other projects

Reply:
Sad, however, yeah, someone who'll assemble something like, will be able to use the protocol for playback :fox_face: